### PR TITLE
feat: remote config

### PR DESCRIPTION
**Pull request by capJavert**
- add remote config support to api over gb feature flags https://app.growthbook.io/features/api-config
- not removing values from env/pulumi yet until we verify deployment on production as well
- currently using for:
  - vordr words
  - vordr ips
  - email domains

### CODE REVIEW

File: bin/cli.ts [hunk 1]
```diff
@@ -6,8 +6,11 @@ import background from '../src/background';
 import temporal from '../src/temporal/notifications';
 import cron from '../src/cron';
 import personalizedDigest from '../src/commands/personalizedDigest';
+import { remoteConfig } from '../src/remoteConfig';
 
 async function run(positionals: string[]) {
+  await remoteConfig.init();
```
Comment (capJavert) [inline bin/cli.ts:12]: this preloads the config for the first time

File: src/remoteConfig.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  private readonly configKey = 'api-config';
+
+  async init(): Promise<void> {
+    this.gb = new GrowthBook({
+      apiHost: 'https://cdn.growthbook.io',
+      clientKey: process.env.GROWTHBOOK_CLIENT_KEY,
+    });
+
+    await this.gb.init({
+      streaming: true,
```
Comment (capJavert) [inline src/remoteConfig.ts:22]: enables streaming of changes, tested it works

File: .env [hunk 3]
```diff
@@ -46,6 +46,7 @@ MEILI_TOKEN=dev
 
 EXPERIMENTATION_KEY=6FA4C8D3DD2400B9D614BB7A229FD69B
 #GROWTHBOOK_CLIENT_KEY=override_me_in_env.development
+#GROWTHBOOK_API_CONFIG_CLIENT_KEY=override_me_in_env.development
```
Comment (capJavert) [inline .env:49]: in dev config will always be `{}`, if something is needed to test gb key can be added as with normal feature flags/exp

File: .infra/Pulumi.prod.yaml [hunk 4]
```diff
@@ -70,6 +70,8 @@ config:
       secure: AAABADH0xT7H6UeMlZG0fUEGjohQ1dvwcbIgJ3WMI5wfqScub3Qp3nvUDhska0DunbFmvj7IrtH07noRL1lCYg==
     flagsmithKey: dontdelete
     flagsmithWebhookSecret: dontdelete
+    growthbookApiConfigClientKey:
```
Comment (capJavert) [inline .infra/Pulumi.prod.yaml:73]: new client key was needed to access different environment that holds api config

File: src/common/utils.ts [hunk 5]
```diff
@@ -165,15 +166,10 @@ export function isNumber(value: string | number): boolean {
   return !isNaN(Number(value.toString()));
 }
 
-const ignoredWorkEmailDomains =
-  process.env.IGNORED_WORK_EMAIL_DOMAINS?.split(',')
-    .filter(Boolean)
-    .map((domain) => domain.toLowerCase()) || [];
```
Comment (capJavert) [inline src/common/utils.ts:171]: no need to map because in config this is now array

**Review by omBratteng — APPROVED** [review]
Haven't tested, but lgtm

**Review by rebelchris — APPROVED** [review]
Looks good to me :) (and seen it in action)

File: __tests__/setup.ts [hunk 6]
```diff
@@ -12,7 +13,21 @@ jest.mock('../src/growthbook', () => ({
   getEncryptedFeatures: jest.fn(),
 }));
 
+jest.mock('../src/remoteConfig', () => ({
```
Comment (idoshamun) [inline __tests__/setup.ts:16]: Nice!

**Review by idoshamun — APPROVED** [review]
Amazing!!

One note for the future, removing env var from pulumi is not as easy as you may think. It causes deployment failure. Something we will need to address as part of pulumi common somehow

**Pull request merged**